sonoff/xsns_48_chirp.ino

```diff
@@ -0,0 +1,484 @@
+uint8_t    chirp_found_sensors  = 0;    // number of found sensors
+
+char       chirp_name[7];
+uint8_t    chirp_next_job       = 0;    //0=reset, 1=auto-wake, 2=moisture+temperature, 3=light, 4 = pause; 5 = TELE done
+uint32_t   chirp_timeout_count  = 0;    //is handled every second, so value is equal to seconds (it is a slow sensor)
+
+#pragma pack(1)
+struct ChirpSensor_t{
+    uint16_t   moisture = 0;      // shall hold post-processed data, if implemented
+    uint16_t   light = 0;         // light level, maybe already postprocessed depending on the firmware
+    int16_t    temperature= 0;    // temperature in degrees CELSIUS * 10
+    uint8_t    version = 0;       // firmware-version
+    uint8_t    address:7;         // we need only 7bit so...
+    uint8_t    explicitSleep:1;   // there is a free bit to play with ;)
+};
+#pragma pack()
+
+ChirpSensor_t chirp_sensor[CHIRP_MAX_SENSOR_COUNT];       // should be 8 bytes per sensor slot
+
+/********************************************************************************************/
+
+void ChirpReset(uint8_t addr) {
+    I2cWriteReg(addr, CHIRP_RESET);
+}
+
+/********************************************************************************************/
+
+void ChirpResetAll(void) {
+    for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+      if (chirp_sensor[i].version) { 
+        ChirpReset(chirp_sensor[i].address);
+        }
+    }
+}
+/********************************************************************************************/
+
+void ChirpClockSet() { // set I2C for this slow sensor
+    Wire.setClockStretchLimit(4000);
+    Wire.setClock(50000);
+}
+
+/********************************************************************************************/
+
+void ChirpSleep(uint8_t addr) {
+    I2cWriteReg(addr, CHIRP_SLEEP);
+}
+
+/********************************************************************************************/
+
+// void ChirpSleepAll(void) {
+//     for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+//       if (chirp_sensor[i].version) { 
+//         ChirpSleep(chirp_sensor[i].address);
+//         }
+//     }
+// }
+
+// /********************************************************************************************/
+
+// void ChirpAutoWakeAll(void) {
+//     for (uint8_t i = 0; i < chirp_found_sensors; i++) {
+//       if (chirp_sensor[i].version && !chirp_sensor[i].explicitSleep) { 
+//         ChirpReadVersion(chirp_sensor[i].address);
+//         }
+//     }
+// }
+
+/********************************************************************************************/
+
+void ChirpSelect(uint8_t sensor) {
+  if(sensor < chirp_found_sensors) { //TODO: show some infos
+    chirp_current = sensor;
+    AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: Sensor %u now active.", chirp_current);
+  }
+  if (sensor == 255) {
+    AddLog_P2(LOG_LEVEL_DEBUG, "CHIRP: Sensor %u active at address 0x%x.", chirp_current, chirp_sensor[chirp_current].address);
```

Comment:
You can use PSTR() to force text in PROGMEM, like:

`AddLog_P2(LOG_LEVEL_DEBUG, PSTR("CHIRP: Sensor %u now active."), chirp_current);`